lib/extractor/index.ts

```diff
@@ -1,28 +1,56 @@
-import { extractDockerArchive } from "./layer";
-import { DockerArchiveManifest, ExtractAction, ExtractedLayers } from "./types";
+import { ImageType } from "../types";
+import * as dockerExtractor from "./docker-archive";
+import * as ociExtractor from "./oci-archive";
+import { ExtractAction, ExtractedLayers, ExtractionResult } from "./types";
 
 /**
- * Given a path on the file system to a docker-archive, open it up to inspect the layers
+ * Given a path on the file system to a image archive, open it up to inspect the layers
  * and look for specific files. File content can be transformed with a custom callback function if needed.
- * @param fileSystemPath Path to an existing docker-archive.
+ * @param fileSystemPath Path to an existing archive.
  * @param extractActions This denotes a file pattern to look for and how to transform the file if it is found.
  * By default the file is returned raw if no processing is desired.
  */
-export async function getDockerArchiveLayersAndManifest(
+export async function getArchiveLayersAndManifest(
+  imageType: ImageType,
   fileSystemPath: string,
   extractActions: ExtractAction[],
-): Promise<{
-  layers: ExtractedLayers;
-  manifest: DockerArchiveManifest;
-}> {
-  const dockerArchive = await extractDockerArchive(
-    fileSystemPath,
-    extractActions,
-  );
+): Promise<ExtractionResult> {
+  switch (imageType) {
```

Comment:
[Minor point]
I think we could avoid some duplicated code here if we created `extractor` constant. What do you think?

```ts
...
  const extractor;
  switch (imageType) {
     case ImageType.OciArchive:
        extractor = ociExtractor;
     default:
        extractor = dockerExtractor;
  }

  const archive = await extractor.extractArchive(
     fileSystemPath,
     extractActions,
   );

   return {
       imageId: extractor.getImageIdFromManifest(archive.manifest),
       manifestLayers: extractor.getManifestLayers(
           archive.manifest,
       ),
       extractedLayers: flattenExtractedLayersArray(archive.layers),
   };
```